### ISSUE TITLE
Reconstruction Precision Problem

### ISSUE THREAD

**Issue by Franciszzj** [YadiraF/PRNet#56]
Hi,

Thanks for your code, and you did a great job!

When I reproduce your work, I found the model trained by me is not as precision as the model trained by you. For example, the following contrast figure.
![default](https://user-images.githubusercontent.com/16440889/43936246-55f80028-9c8a-11e8-81f0-4fd7fd7441ed.jpg)
The left is the output of your model, and the right is mine.

I find that the facial contour/texture details of the model I trained are very imprecise. Do you need other skills to train the model? Are there any other ways to improve performance other than those mentioned in the paper?

I look forward to your reply very much. Thank you.

**Comment by jhgfkdj** [comment]
Hi I wonder if you wrote the training code by yourself ? and you used your own dataset to fine-tune author's model?

**Comment by Franciszzj** [comment]
@jhgfkdj I reproduce PRNet use MXNet, and use 300W-LP to train my model. I don't fine-tune the author's model because it is based on TensorFlow.

**Comment by Franciszzj** [comment]
Hi, all.
I solved this problem. Just not use the *.jpg or *.png label form to save label, while save position map label as npy, and read npy during training.
Make sure no precision loss on label!

**Comment by jhgfkdj** [comment]
@Franciszzj Well I think the author did the same as you shared, see [here](https://github.com/YadiraF/face3d/blob/master/examples/8_generate_posmap_300WLP.py)

**Comment by Franciszzj** [comment]
@jhgfkdj Yes, but for MXNet, we often package data into rec files to speed up data io.

**Comment by chenyihang1993** [comment]
@Franciszzj Could you share the  PRNet code reproduced by MXNet? I really need it！

**Comment by KevinLee752** [comment]
@Franciszzj Hi! I did train the model with label saved as npy, but I still get an unprecision face model somehow like  the result you posted at first time. Would you please give me some advice? By the way, I didn't do the data augmentation like the paper said, and I'm wondering if it is the problem.

**Comment by franciszzj** [comment]
> @Franciszzj Hi! I did train the model with label saved as npy, but I still get an unprecision face model somehow like the result you posted at first time. Would you please give me some advice? By the way, I didn't do the data augmentation like the paper said, and I'm wondering if it is the problem.

@KevinLee752 YES! Make sure you use the same data aug and optimizer strategies as paper did. Hope you get better results.

**Comment by vitahsu** [comment]
@KevinLee752  @Franciszzj 
sorry to bother you guys!
I'm reproducing model now and I face some problems.
When training model, should we do normalization on training input data(original 2D image)?
Did you guys reproduce model on tensorflow?
If yes, what hyper parameter you guys set(learning rate, epoch, batch)? 
Now I'm facing that result have same position on all vertex, I cannot find what problem to model.

.obj like this:
v 231.068704044 172.688475031 177.332809448 0.737254901961 0.411764705882 0.286274509804
v 231.090035233 172.685373009 177.320831299 0.737254901961 0.411764705882 0.286274509804
v 231.063706342 172.687900582 177.333007812 0.737254901961 0.411764705882 0.286274509804
v 231.086990656 172.688283548 177.320907593 0.737254901961 0.411764705882 0.286274509804
v 231.059493719 172.696555607 177.336380005 0.737254901961 0.411764705882 0.286274509804
v 231.085592831 172.686981464 177.320404053 0.737254901961 0.411764705882 0.286274509804
v 231.06372549 172.688666513 177.331939697 0.737254901961 0.411764705882 0.286274509804

Did anyone face this problem before?
Please give me some advice ><.
Thank you so much!
